MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/dds/PolygonSelectToggleActivity.kt

```diff
@@ -0,0 +1,290 @@
+        featureCollection = collection
+
+        // Create a GeoJsonSource and add it to the map.
+        geoJsonSource = GeoJsonSource(NEIGHBORHOOD_POLYGON_SOURCE_ID, featureCollection)
+
+        mapboxMap?.style?.addSource(geoJsonSource!!)
+
+        // Create a FillLayer which will show a background of all neighborhoods on the map.
+        val neighborhoodPolygonBaseFillLayer = FillLayer(BASE_NEIGHBORHOOD_FILL_LAYER_ID,
+                NEIGHBORHOOD_POLYGON_SOURCE_ID)
+        neighborhoodPolygonBaseFillLayer.withProperties(
+                fillOpacity(.2f))
+
+        // Add neighborhood layer below the city label layer
+        mapboxMap?.style?.addLayerBelow(neighborhoodPolygonBaseFillLayer, "settlement-label")
+
+        // Add a FillLayer which will only show "selected" features/polygons/neighborhoods.
+        val neighborhoodPolygonColoredFillLayer = FillLayer(COLORED_FILL_LAYER_ID,
+                NEIGHBORHOOD_POLYGON_SOURCE_ID)
+        neighborhoodPolygonColoredFillLayer.withProperties(
+
+                // Use data-driven styling to use each Feature's color property, which was set in
+                // `onPostExecute()` of the data loading AsyncTask.
+                fillColor(toColor(get(PROPERTY_FILL_COLOR))),
+                fillOpacity(.95f))
+
+        // Add a filter to ensure that only "selected" features/polygons/neighborhoods are displayed.
+        neighborhoodPolygonColoredFillLayer.withFilter(eq((get(PROPERTY_SELECTED)), literal(true)))
+        mapboxMap?.style?.addLayerBelow(neighborhoodPolygonColoredFillLayer, "settlement-label")
+
+        // Add LineLayer to outline the various neighborhoods' areas
+        val neighborhoodOutlineLineLayer = LineLayer(LINE_LAYER_ID, NEIGHBORHOOD_POLYGON_SOURCE_ID)
+        neighborhoodOutlineLineLayer.withProperties(
+                lineColor(Color.GRAY),
+                lineWidth(2.2f))
+        mapboxMap?.style?.addLayerBelow(neighborhoodOutlineLineLayer, "settlement-label")
+    }
+
+    /**
+     * This method handles click events for SymbolLayer symbols.
+     * When a SymbolLayer icon is clicked, we adjust that feature to the selected state.
+     * @param screenPoint the point on screen clicked
+     */
+    private fun handleClickIcon(screenPoint: PointF): Boolean {
+        val features = mapboxMap?.queryRenderedFeatures(screenPoint, BASE_NEIGHBORHOOD_FILL_LAYER_ID)
+        if (features?.isEmpty() == false) {
+            val name = features[0].getStringProperty(NEIGHBORHOOD_NAME_PROPERTY)
+            val featureList = featureCollection?.features()
+            for (i in featureList!!.indices) {
+                if (featureList[i].getStringProperty(NEIGHBORHOOD_NAME_PROPERTY) == name) {
+                    if (featureSelectStatus(i)) {
+                        setFeatureSelectState(featureList.get(i), false)
+                    } else {
+                        setSelected(i)
+                    }
+                }
+            }
```

Comment:
there might be a cleaner Koltin way of handling this iteration as a for i loop

Reply:
replaced with

```kotlin
            (0 until featureList!!.size).iterator().forEach {
                if (featureList[it].getStringProperty(NEIGHBORHOOD_NAME_PROPERTY) == name) {
                    if (featureSelectStatus(it)) {
                        setFeatureSelectState(featureList.get(it), false)
                    } else {
                        setSelected(it)
                    }
                }
            }
```